src/oca_pre_commit_hooks/check_unused_python_file.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+import ast
+import functools
+import os
+from typing import List, Set, Union
+
+
+@functools.lru_cache()
+def get_imported_files(dirname: str) -> Set[str]:
+    imported_files = set()
+    init_file = os.path.join(dirname, "__init__.py")
+    if os.path.exists(init_file):
+        with open(init_file, encoding="utf-8") as init_fd:
+            init_ast = ast.parse(init_fd.read())
+
+        if isinstance(init_ast.body, list):
+            for module in filter(lambda elem: isinstance(elem, ast.ImportFrom), init_ast.body):
+                for name in module.names:
+                    imported_files.add(name.name)
+
+    return imported_files
+
+
+def check_unused_python_file(filenames: List[str]) -> int:
+    status = 0
+    for filename in filenames:
+        if os.path.basename(filename)[:-3] not in get_imported_files(os.path.dirname(filename)):
+            print(f"{filename}: not imported")
+            status = -1
+
+    return status
+
+
+def main(argv: Union[List[str], None] = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filenames", nargs="*")
```

Comment:
Could you consider to add an extra parameter to change the exit_status in order to be able to enable it but without fails, please?

Similar to:
https://github.com/PyCQA/bandit/blob/d9fe642e01866e460454641bcd14b9de9d2b1478/bandit/cli/main.py#L363-L369

Reply:
I think the behavior for all these hooks should be standard. I understand we are the main users for it and changes like this can blow up our whole CI, but if we separate concerns, I think that is not a concern of this repo, but instead of the users (us).

There is already a "native" option for users to run hooks but make them optional, I think we could use that instead of changing the code.

https://stackoverflow.com/questions/59730765/is-it-possible-to-run-mypy-pre-commit-without-making-it-fail